### PR TITLE
[wasm] fix missing managed stack trace on managed exceptions marshaled to JS

### PR DESCRIPTION
**Pull request by pavelsavara**
When we display managed exception marshaled to JS as `ManagedError` via `toString` we only print:
```
ManagedError(gc_handle: 214)
```

The fix makes it to print both managed and JS stack trace
```
     at System.Runtime.InteropServices.JavaScript.Tests.JavaScriptTestHelper.ThrowFromJSExport(String message) in JavaScriptTestHelper.cs:line 41
     at System.Runtime.InteropServices.JavaScript.Tests.JavaScriptTestHelper.__Wrapper_ThrowFromJSExport_849163347(JSMarshalerArgument* __arguments_buffer) in JSExports.g.cs:line 50
  Error: -t-e-s-t-
      at marshal_exception_to_js (dotnet.js:2803:22)
      at invoke_method_and_handle_exception (dotnet.js:5030:19)
      at bound_fn_1R (dotnet.js:4964:17)
      at Module.catch1toString (JavaScriptTestHelper.mjs:117:9)
      at bound_fn_2R (dotnet.js:4696:35)
      at Object.mono_wasm_invoke_import (dotnet.js:4747:9)
      at _mono_wasm_invoke_import (dotnet.js:16384:82)
      at do_icall (do_icall (wasm://wasm/02e6dfee:wasm-function[308]:0x1dbed))
      at do_icall_wrapper (do_icall_wrapper (wasm://wasm/02e6dfee:wasm-function[285]:0x1cd26))
      at interp_exec_method (interp_exec_method (wasm://wasm/02e6dfee:wasm-function[235]:0xc4ca))
      at interp_runtime_invoke (interp_runtime_invoke (wasm://wasm/02e6dfee:wasm-function[234]:0xb1bd))
      at mono_jit_runtime_invoke (mono_jit_runtime_invoke (wasm://wasm/02e6dfee:wasm-function[15606]:0x27f8d3))
      at do_runtime_invoke (do_runtime_invoke (wasm://wasm/02e6dfee:wasm-function[3302]:0xe9376))
      at mono_runtime_invoke_checked (mono_runtime_invoke_checked (wasm://wasm/02e6dfee:wasm-function[3300]:0xe9251))
      at mono_runtime_try_invoke_span (mono_runtime_try_invoke_span (wasm://wasm/02e6dfee:wasm-function[3449]:0xf1d9d))
      at mono_runtime_invoke_span_checked (mono_runtime_invoke_span_checked (wasm://wasm/02e6dfee:wasm-function[3450]:0xf21b8))
      at ves_icall_InternalInvoke (ves_icall_InternalInvoke (wasm://wasm/02e6dfee:wasm-function[1974]:0xa0a88))
      at ves_icall_InternalInvoke_raw (ves_icall_InternalInvoke_raw (wasm://wasm/02e6dfee:wasm-function[2197]:0xaeca0))
      at do_icall (do_icall (wasm://wasm/02e6dfee:wasm-function[308]:0x1dc71))
      at do_icall_wrapper (do_icall_wrapper (wasm://wasm/02e6dfee:wasm-function[285]:0x1cd26))
      at interp_exec_method (interp_exec_method (wasm://wasm/02e6dfee:wasm-function[235]:0xc4ca))
      at interp_runtime_invoke (interp_runtime_invoke (wasm://wasm/02e6dfee:wasm-function[234]:0xb1bd))
      at mono_jit_runtime_invoke (mono_jit_runtime_invoke (wasm://wasm/02e6dfee:wasm-function[15606]:0x27f8d3))
      at do_runtime_invoke (do_runtime_invoke (wasm://wasm/02e6dfee:wasm-function[3302]:0xe9376))
      at mono_runtime_try_invoke (mono_runtime_try_invoke (wasm://wasm/02e6dfee:wasm-function[3316]:0xea048))
      at mono_runtime_invoke (mono_runtime_invoke (wasm://wasm/02e6dfee:wasm-function[3382]:0xee01d))
      at mono_wasm_invoke_method_bound (mono_wasm_invoke_method_bound (wasm://wasm/02e6dfee:wasm-function[129]:0x8efa))
      at Module._mono_wasm_invoke_method_bound [as mono_wasm_invoke_method_bound] (dotnet.js:17161:149)
      at invoke_method_and_handle_exception (dotnet.js:5025:42)
      at runtimeHelpers.javaScriptExports.complete_task (dotnet.js:5185:17)
      at dotnet.js:3206:46
```

### CODE REVIEW

**Comment by pavelsavara** [comment]
Issues are https://github.com/dotnet/runtime/issues/75687 and https://github.com/dotnet/runtime/issues/75684

File: src/mono/wasm/runtime/logging.ts [hunk 1]
```diff
@@ -60,10 +61,11 @@ export function mono_wasm_symbolicate_string(message: string): string {
     }
 }
 
-export function mono_wasm_stringify_as_error_with_stack(err: Error | string): string {
+export function mono_wasm_stringify_as_error_with_stack(err: Error | ManagedError | string): string {
     let errObj: any = err;
-    if (!(err instanceof Error))
-        errObj = new Error(err);
+    if (!(errObj instanceof Error) && !(errObj instanceof ManagedError)) {
```
Comment (kg) [inline src/mono/wasm/runtime/logging.ts:66]: Is managederror not derived from error?

File: src/mono/wasm/runtime/marshal.ts [hunk 2]
```diff
@@ -335,7 +341,7 @@ export class ManagedError extends Error implements IDisposable {
     }
 
     toString(): string {
-        return `ManagedError(gc_handle: ${(<any>this)[js_owned_gc_handle_symbol]})`;
+        return this.managedStack || this.message;
```
Comment (kg) [inline src/mono/wasm/runtime/marshal.ts:344]: js `.stack` doesn't contain the exception message, is it intended that this is only the stack and not message + stack

**Review by kg — CHANGES_REQUESTED** [review]
Looks fine other than the ManagedError inheritance question and toString

File: src/mono/wasm/runtime/logging.ts (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/logging.ts]: I have not realized that `instanceof` follows prototype chain, thanks!

File: src/mono/wasm/runtime/marshal.ts (hunk 2 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/marshal.ts:344]: It does on chrome

image

Comment (pavelsavara) [reply src/mono/wasm/runtime/marshal.ts]: But I will stick to stack without message

**Review by kg — APPROVED** [review]

**Pull request merged**